Frameworks/include/CGIWICBitmap.h

```diff
@@ -187,17 +187,36 @@ class CGIWICBitmap : public Microsoft::WRL::RuntimeClass<Microsoft::WRL::Runtime
                                          _Out_writes_all_(cbBufferSize) BYTE* buffer) {
         RETURN_HR_IF_NULL(E_POINTER, buffer);
 
+        const WICRect fullRect = { 0, 0, m_width, m_height };
         Microsoft::WRL::ComPtr<IWICBitmapLock> lock;
-        RETURN_IF_FAILED(Lock(copyRect, 0, &lock));
+        // TODO #1379: Support sub-regional locking.
+        RETURN_IF_FAILED(Lock(&fullRect, 0, &lock));
 
+        if (!copyRect) {
+            copyRect = &fullRect;
+        }
+
+        UINT sourceDataStride;
         UINT sourceDataSize;
         BYTE* sourceData;
+        RETURN_IF_FAILED(lock->GetStride(&sourceDataStride));
         RETURN_IF_FAILED(lock->GetDataPointer(&sourceDataSize, &sourceData));
 
-        RETURN_HR_IF(E_INVALIDARG, sourceDataSize > bufferSize);
-
-        // TODO #1379 - should support regional copying.
-        RETURN_HR_IF(E_UNEXPECTED, memcpy_s(buffer, bufferSize, sourceData, sourceDataSize) != 0);
+        if (copyRect->X == 0 && copyRect->Y == 0 && copyRect->Width == m_width && copyRect->Height == m_height) {
+            RETURN_HR_IF(E_INVALIDARG, sourceDataSize > bufferSize);
+            RETURN_HR_IF(E_UNEXPECTED, memcpy_s(buffer, bufferSize, sourceData, sourceDataSize) != 0);
+        } else {
+            // Once we support sub-regional locking we can fix this stride copier.
+            size_t bytesPerPixel = sourceDataStride / fullRect.Width;
```

Comment:
It can. That'd be a <8bpp image, and we should bomb out for that ;P

---
In reply to: [92295930](https://github.com/Microsoft/WinObjC/pull/1529#discussion_r92295930) [](ancestors = 92295930)